tensorrt_llm/_torch/models/modeling_llama.py

```diff
@@ -743,7 +758,33 @@ def forward(
     def load_weights(self, weights: Dict):
         new_weights = {}
         for k, v in weights.items():
-            new_k = "model." + k if 'lm_head' not in k and "embed_tokens" not in k else k
+            new_k = "model." + k if 'lm_head' not in k else k
             new_weights[new_k] = v
 
         super().load_weights(new_weights)
+
+    def load_weights_from_target_model(self,
+                                       target_model: torch.nn.Module) -> None:
+        if self.model.embed_tokens is None:
+            self.model.embed_tokens = target_model.model.embed_tokens
+
+    # TODO: should input/position IDs be included in this? Keeping it implicit
+    # for now since the shapes/dtypes are the same across all models we have.
+    def get_warmup_extra_inputs(self, batch_size: int,
+                                num_tokens: int) -> Dict[str, Any]:
+
+        hidden_states = torch.empty(batch_size * num_tokens,
+                                    self.model.hidden_size_in,
+                                    dtype=self.model.dtype,
+                                    device='cuda')
+
+        return {'hidden_states': hidden_states}
+
+    def apply_eagle3_fc(self, hidden_states: torch.Tensor) -> torch.Tensor:
```

Comment:
Can you add some comments on why this has to be a separate function and where it's called? Similar to the comments you add to inside `_prepare_draft_tokens`.